### PR TITLE
Minor improvements for ListThresholdsForCreate

### LINKED ISSUE #8648: vocms0256 has created > 3 million jobs (!)
There is something weird going on with the production system.
According to grafana, today at 15:05 CERN time there was a spike of requests in ReqMgr2 (jumping from ~4k to ~17k requests in the system), where ~6k of them were in acquired status. (I bet it was a monit glitch, since it's back to normal after 15:15).

Then looking at the agents, it seems vocms0256 managed to acquire and create more than 3M jobs
```
2018-06-11 15:57:47,223:139947075262208:INFO:JobSubmitterPoller:Found 3375907 new jobs to be submitted.
```

having a quick look at wmstats, some of those workflows are
pdmvserv_task_SMP-RunIISummer15wmLHEGS-00224__v1_T_180416_194001_5265
pdmvserv_task_SMP-RunIISummer15wmLHEGS-00225__v1_T_180416_194002_549
pdmvserv_task_SMP-RunIISummer15wmLHEGS-00226__v1_T_180416_194018_271
pdmvserv_task_SMP-PhaseIISummer17GenOnly-00022__v1_T_180607_163200_4994

all MC from scratch AFAICT.

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #8648 (eventually)

I just wanted to see whether I really understood this DAO and to make sure my changes don't break anything.
Next step is to look at Acquired LQE (local workqueue db), which I think we can actually look at wmbs subscriptions somehow.

### CODE REVIEW

**Comment by amaltaro** [comment]
The failing unit test is legit, even though I'm not sure the use case is valid...
The current behavior is that subscriptions without a site white/black list will use the job location assigned to count a pending_job, while the code with these changes will just disconsider any subscriptions without white/black locations (valid is Null).

@ticoann do you know whether we can have a wmbs_subscription where the corresponding wmbs_subscription_validation.valid is Null? I guess not, but...

**Comment by ticoann** [comment]
Thanks Alan, that is smart. The one concern is when there is very slow job creation as you observed. Agent will stop pulling the work until every subscription is finished.  so it won't pull the enough work and work will go to other agents. (However it might be a right behavior if JobCreator has some problems).

**Comment by amaltaro** [comment]
I've done the WorkQueueManager vs JobCreator tests and it works as expected. Not pulling any new work from GQ while there are still subscriptions not handled by JobCreator.

I'm going to try to better understand those queries now and check whether my SQL changes cause more benefit than drawbacks. If we keep that, I still have to update one unit test.

File: src/python/WMCore/ResourceControl/MySQL/ListThresholdsForCreate.py [hunk 1]
```diff
@@ -71,11 +71,7 @@ class ListThresholdsForCreate(DBFormatter):
                             WHERE wmbs_job.location IS NULL AND
                                   wmbs_job_state.name != 'killed' AND
                                   wmbs_job_state.name != 'cleanout' AND
-                                  (wsv.valid = 1 OR
-                                   (wsv.valid IS NULL AND NOT EXISTS
```
Comment (amaltaro) [inline src/python/WMCore/ResourceControl/MySQL/ListThresholdsForCreate.py]: I had a look at vocms0254 db and I cannot find anything for `wsv.valid IS NULL AND...`.
Still, running the original sub-query (unassigned_jobs) and this one, I get around 246k rows while the latter gives me 245k rows.

Comment (ticoann) [reply src/python/WMCore/ResourceControl/MySQL/ListThresholdsForCreate.py]: you can find  wsv.valid is NULL from the table but only when you do the left outer join. Were you checking that? I am still trying to understand original query

**Comment by amaltaro** [comment]
Seangchan, as we discussed today, I moved the 2nd commit to https://github.com/dmwm/WMCore/pull/8667 and I/we will work on it for the new agent only. I'll bug Kate after CHEP too.

**Comment by amaltaro** [comment]
test this please

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
The failing unit test was explained here:
https://github.com/dmwm/WMCore/pull/8663#issuecomment-400324402

I guess that's not a valid case though (having a subscription without any valid location), so I'll just update the unit test.

File: src/python/WMCore/ResourceControl/MySQL/ListThresholdsForCreate.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/ResourceControl/MySQL/ListThresholdsForCreate.py]: I'm not sure I got the correct query. However, looking at the code, I see we do not have any update action for wmbs_subscription_validation. So either a subscription has a site white/black list or it doesnt:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/WMBSHelper.py#L294

That means, removing this part of the code _should_ indeed be fine

**Comment by amaltaro** [comment]
I was wrong, merge subscriptions don't have an entry in wmbs_sub_validation (so no sites white/black listed). See
```
MariaDB [wmagent]> select count(distinct wmbs_subscription.id) from wmbs_sub_types
    ->   inner join wmbs_subscription on wmbs_subscription.subtype = wmbs_sub_types.id
    -> where name = 'Merge' and wmbs_subscription.id not in 
    ->   (select distinct subscription_id from wmbs_subscription_validation);
+--------------------------------------+
| count(distinct wmbs_subscription.id) |
+--------------------------------------+
|                                  199 |
+--------------------------------------+
1 row in set (0.00 sec)

MariaDB [wmagent]> select count(distinct wmbs_subscription.id) from wmbs_sub_types
    ->   inner join wmbs_subscription on wmbs_subscription.subtype = wmbs_sub_types.id
    -> where name = 'Merge' and wmbs_subscription.id in 
    ->   (select distinct subscription_id from wmbs_subscription_validation);
+--------------------------------------+
| count(distinct wmbs_subscription.id) |
+--------------------------------------+
|                                    0 |
+--------------------------------------+
1 row in set (0.00 sec)
```

I did however a very very quick performance test for the remaining code changes (not the SQL) and it seems to be around 10% faster (and cleaner). So I'm going to roll back the SQL changes and we can go ahead with the formatting changes only. Does it sound good, Seangchan?

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
I give up trying to understand what's wrong with this patch and the failing unit test. Even without a SQL change, it would give us a performance boost, but I'm closing this one.

**Pull request closed without merging**